internal/multierror/multierror.go

```diff
@@ -0,0 +1,44 @@
+package multierror
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Join is a drop-in replacement for errors.Join with better formatting.
+func Join(errs ...error) error {
+	n := 0
+	for _, err := range errs {
+		if err != nil {
+			n++
+		}
+	}
+	if n == 0 {
+		return nil
+	}
+	e := &joinError{
+		errs: make([]error, 0, n),
+	}
+	for _, err := range errs {
+		if err != nil {
+			e.errs = append(e.errs, err)
+		}
+	}
+	return e
+}
+
+type joinError struct {
+	errs []error
+}
+
+func (e *joinError) Error() string {
+	stringErrs := make([]string, 0, len(e.errs))
```

Comment:
Wondering if we should take "single" errors into account; that way a "single-error-multierror" doesn't create a bullet list with one item;

```suggestion
func (e *joinError) Error() string {
	if len(e.errs) == 1 {
		return strings.TrimSpace(e.errs[0].Error())
	}
```